src/components/GearForm.tsx

```diff
@@ -0,0 +1,307 @@
+import React, { FunctionComponent, useState } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import {
+  List,
+  AppBar,
+  Toolbar,
+  IconButton,
+  Typography,
+  Button,
+  Dialog
+} from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+import FilterDrawer from "./FilterDrawer";
+import Database from "./Database.js";
+import FilterListIcon from "@material-ui/icons/FilterList";
+import SearchIcon from "@material-ui/icons/Search";
+import GearList from "./GearList";
+import { CalendarUIProps, CalendarAction } from "../calendar/types";
+import { makeTransition } from "./Transition";
+import Gear from "../resources/Gear";
+import Filter from "../resources/Filter";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1
+  },
+  menuButton: {
+    marginRight: theme.spacing(2)
+  },
+  title: {
+    flexGrow: 1,
+    color: "white",
+    textAlign: "center",
+    fontSize: "18px"
+  }
+}));
+
+const tempFilters = [
+  {
+    name: "",
+    toggle: false
+  }
+];
+const initialFilters: Filter[] = tempFilters;
+
+const transition = makeTransition("up");
+
+const GearForm: FunctionComponent<CalendarUIProps> = ({ dispatch, state }) => {
+  const [drawerIsOpen, setDrawerIsOpen] = useState(false);
+  const [matchAny, setMatchAny] = useState(false);
+  const classes = useStyles();
+  const gear: Gear[] = state.gear;
+
+  const viewFilters: Filter[] = [];
+  const [filters, setFilters] = useState(initialFilters);
+  const [searchString, setSearchString] = useState("");
+
+  function checkExists(tag: string): boolean {
+    let hasMatch = false;
+    for (let index = 0; index < filters.length; ++index) {
+      if (filters[index].name === tag) {
+        hasMatch = true;
+      }
+    }
+    return hasMatch;
+  }
+
+  function pushArray(tag: string): void {
+    const tempArray = filters;
+    const filter = new Filter({
+      name: tag,
+      toggle: false
+    });
+    tempArray.push(filter);
+    setFilters(tempArray);
+  }
+
+  function checkViewExists(tag: string): boolean {
+    let hasMatch = false;
+    for (let index = 0; index < viewFilters.length; ++index) {
+      if (viewFilters[index].name === tag) {
+        hasMatch = true;
+      }
+    }
+    return hasMatch;
+  }
+
+  function pushViewArray(tag: string): void {
+    const viewArray = viewFilters;
+    const filter = filters.find(function (filter) {
+      return filter.name === tag;
+    });
+    if (filter) {
+      viewArray.push(filter);
+    }
+  }
+
+  function cleanTag(tag: string): string {
+    tag = tag.trim();
+    const words = tag.split(" ");
+    tag = "";
+    for (let index = 0; index < words.length; ++index) {
+      const word = words[index].charAt(0).toUpperCase() + words[index].slice(1);
+      tag = tag + word + " ";
+    }
+    return tag.trim();
+  }
```

Comment:
What's going on here?  A comment to explain the intent would be helpful.
It looks like you are trying to transform
```
"a bunch of tags" => "A Bunch Of Tags"
```
but I can't figure out why you want to do this.